### PR TITLE
SDC parsing support

### PR DESCRIPTION
**Pull request by rowanG077**
The coming 2 months I have a lot of time to work on nextpnr. I want to make the timing changes I make to be small so I will make small PRs to get them merged asap. 

First step is to get SDC support merged with clock frequency support.

### CODE REVIEW

**Comment by rowanG077** [comment]
I will be adding test cases to: https://github.com/rowanG077/nextpnr-thesis-tests

**Comment by rowanG077** [comment]
clock constraint with frequency is confirmed working.

File: common/kernel/sdc.cc [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+        if (type == ENTITY_PIN) {
+            CellInfo *cell = nullptr;
+            if (ctx->cells.count(name)) {
+                cell = ctx->cells.at(name).get();
+            } else {
+                const std::string &n = name.str(ctx);
+                auto pos = n.rfind('.');
+                if (pos == std::string::npos)
+                    return nullptr;
+                // remove one hierarchy layer due to Radiant weirdness around PLLs etc
```
Comment (gatecat) [inline common/kernel/sdc.cc:72]: this is a nexus-specific workaround that I don't think needs to be kept in the generic code

File: common/kernel/sdc.cc [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+#include "log.h"
+#include "nextpnr.h"
+
+#include <algorithm>
+#include <iterator>
+
+NEXTPNR_NAMESPACE_BEGIN
+
+struct TclEntity
```
Comment (gatecat) [inline common/kernel/sdc.cc:30]: to avoid a collision with arch-specific stuff, can this be called SdcEntity instead?

File: common/kernel/sdc.cc [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+            }
+        }
+
+        if (from == nullptr) {
+            log_error("-from is required for set_false_path (line %d)\n", lineno);
+        } else if (to == nullptr) {
+            log_error("-to is required for set_false_path (line %d)\n", lineno);
+        }
+
+        log_info("set_false_path from: %s, to: %s\n", from->name.c_str(ctx), to->name.c_str(ctx));
```
Comment (gatecat) [inline common/kernel/sdc.cc:430]: this should probably be a warning that the `set_false_path` is currently being ignored

Comment (rowanG077) [reply common/kernel/sdc.cc]: Fixed

File: common/kernel/sdc.cc (hunk 2 above)
Comment (rowanG077) [reply common/kernel/sdc.cc]: Fixed

**Pull request merged**